src/Agent/NewRelic/Agent/Core/DataTransport/Client/NRHttpClient.cs

```diff
@@ -65,7 +102,9 @@ public override IHttpResponse Send(IHttpRequest request)
                     req.Content.Headers.Add(contentHeader.Key, contentHeader.Value);
                 }
 
-                var response = _httpClientWrapper.SendAsync(req).GetAwaiter().GetResult();
+                Log.Finest($"Request({request.RequestGuid}: Sending");
+                var response = AsyncHelper.RunSync(() => _httpClientWrapper.SendAsync(req));
```

Comment:
We may not need this helper method, because ConfigureAwait(false) seems to help resolve the same problem in a simpler manner.